code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -455,6 +457,96 @@
 	last_piece = NC
 	return TRUE
 
+/obj/item/mecha_parts/mecha_equipment/eng_toolset
+	name = "Engineering toolset"
+	desc = "Equipment for engineering exosuits. Gives a set of good tools."
+	icon_state = "mecha_toolset"
+	item_state = "artistic_toolbox"
+	lefthand_file = 'icons/mob/inhands/equipment/toolbox_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/toolbox_righthand.dmi'
+	hitsound = 'sound/weapons/smash.ogg'
+	drop_sound = 'sound/items/handling/toolbox_drop.ogg'
+	pickup_sound =  'sound/items/handling/toolbox_pickup.ogg'
+	force = 10
+	equip_cooldown = 1.5 SECONDS
+	energy_drain = 100
+	harmful = TRUE
+	var/list/items_list = newlist(/obj/item/screwdriver/cyborg, /obj/item/wrench/cyborg, /obj/item/weldingtool/experimental/mecha,
+		/obj/item/crowbar/cyborg, /obj/item/wirecutters/cyborg, /obj/item/multitool/cyborg) //0.5 toolspeed all
+	var/obj/item/selected_item
+	var/emag_item = /obj/item/kitchen/knife/combat/cyborg/mecha
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/New()
+	..()
+	for(var/obj/item/item as anything in items_list)
+		item.flags |= NODROP
+		item.resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+		item.slot_flags = null
+		item.w_class = WEIGHT_CLASS_HUGE
+		item.materials = null
+		item.tool_enabled = TRUE
+	selected_item = pick(items_list)
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/get_module_equip_info()
+	for(var/obj/item/item as anything in items_list)
+		var/short_name = uppertext(item.name[1])
+		if(item == selected_item)
+			. += "|<b>[short_name]</b> "
+		else
+			. += "|<a href='?src=[UID()];select=[item.UID()]'>[short_name]</a>"
+	. += "|"
+
+/obj/item/mecha_parts/mecha_equipment/eng_toolset/Topic(href,href_list)
+	..()
+	if(href_list["select"])
+		selected_item = locateUID(href_list["select"])
+		occupant_message("Switched to [selected_item]")
+		update_equip_info()
```

Comment:
you need to sanity check this UID. If you get a null href value or the item somehow gets deleted, you need to make sure that the toolset doesn't break.

Reply:
How to properly do that? 
else return?

Reply:
```dm
if(!istype(selected_item)
   return
```

Reply:
Update it, thanks!

Reply:
> ```dm
> if(!istype(selected_item)
>    return
> ```

Not working, i dunno how to do it properly but your variant is not working

Reply:
is it properly indented?

Reply:
Done